src/new/wallet.js

```diff
@@ -0,0 +1,441 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import EventEmitter from 'events';
+import networkInstance from '../network';
+import wallet from '../wallet';
+import { GAP_LIMIT, HATHOR_TOKEN_CONFIG, DEFAULT_SERVER } from '../constants';
+import transaction from '../transaction';
+import tokens from '../tokens';
+import version from '../version';
+import ws from '../WebSocketHandler';
+import storage from '../storage';
+import MemoryStore from '../memory_store';
+import walletApi from '../api/wallet';
+
+/**
+ * This is a Wallet that is supposed to be simple to be used by a third-party app.
+ *
+ * This class handles all the details of syncing, including receiving the same transaction
+ * multiple times from the server. It also keeps the balance of the tokens updated.
+ *
+ * It has the following states:
+ * - CLOSED: When it is disconnected from the server.
+ * - CONNECTING: When it is connecting to the server.
+ * - SYNCING: When it has connected and is syncing the transaction history.
+ * - READY: When it is ready to be used.
+ *
+ * You can subscribe for the following events:
+ * - state: Fired when the state of the Wallet changes.
+ * - new-tx: Fired when a new tx arrives.
+ * - update-tx: Fired when a known tx is updated. Usually, it happens when one of its outputs is spent.
+ * - more-addresses-loaded: Fired when loading the history of transactions. It is fired multiple times,
+ *                          one for each request sent to the server.
+ **/
+class Wallet extends EventEmitter {
+  constructor({ network, server, seed, tokenUid }) {
+    super();
+
+    networkInstance.setNetwork(network);
+    this.network = network;
+
+    this.state = Wallet.CLOSED;
+    this.serverInfo = null;
+
+    this.onConnectionChange = this.onConnectionChange.bind(this);
+    this.handleWebsocketMsg = this.handleWebsocketMsg.bind(this);
+    this.onAddressesLoaded = this.onAddressesLoaded.bind(this);
+
+    this.seed = seed;
+    this.server = server;
+    if (!this.server) {
+      // We set the default server of the lib
+      this.server = DEFAULT_SERVER;
+
+    }
+
+    // tokenUid is optional so we can get the token of the wallet
+    this.token = null;
+    this.tokenUid = tokenUid;
```

Comment:
If the user does not pass one, we could have it use HTR by default. In this case, there would be no exceptions if the user later calls something like `getBalance` without a parameter.

Reply:
I think HTR should be the default no matter what is passed here. Or we should rename it to `defaultToken`.

Reply:
It's the default token uid now